tests/baselines/reference/mappedTypeContextualTypesApplied.errors.txt

```diff
@@ -0,0 +1,38 @@
+tests/cases/compiler/mappedTypeContextualTypesApplied.ts(18,10): error TS7023: 'foo' implicitly has return type 'any' because it does not have a return type annotation and is referenced directly or indirectly in one of its return expressions.
+tests/cases/compiler/mappedTypeContextualTypesApplied.ts(18,15): error TS7006: Parameter 's' implicitly has an 'any' type.
+tests/cases/compiler/mappedTypeContextualTypesApplied.ts(19,10): error TS7023: 'foo' implicitly has return type 'any' because it does not have a return type annotation and is referenced directly or indirectly in one of its return expressions.
+tests/cases/compiler/mappedTypeContextualTypesApplied.ts(19,15): error TS7006: Parameter 's' implicitly has an 'any' type.

```

Comment:
Those are **not** expected. I'm almost certain that all of those would get fixed if this PR would get reviewed and merged: https://github.com/microsoft/TypeScript/pull/52095 . This currently fails exactly because the contextual type involves an intersection and `getTypeOfPropertyOfContextualType` doesn't handle such cases appropriately.